packages/components/nodes/chains/ApiChain/ApiChain.ts

```diff
@@ -0,0 +1,78 @@
+import { ICommonObject, INode, INodeData, INodeParams } from '../../../src/Interface'
+import { APIChain } from 'langchain/chains'
+import { CustomChainHandler, getBaseClasses } from '../../../src/utils'
+import { BaseLanguageModel } from 'langchain/base_language'
+import { Document } from 'langchain/document'
+import { PromptTemplate } from 'langchain/prompts'
+
+class ApiChain_Chains implements INode {
+    label: string
+    name: string
+    type: string
+    icon: string
+    category: string
+    baseClasses: string[]
+    description: string
+    inputs: INodeParams[]
+
+    constructor() {
+        this.label = 'API Chain'
+        this.name = 'apiChain'
+        this.type = 'ApiChain'
+        this.icon = 'apichain.svg'
+        this.category = 'Chains'
+        this.description = 'Chain to run queries against API'
+        this.baseClasses = [this.type, ...getBaseClasses(APIChain)]
+        this.inputs = [
+            {
+                label: 'Language Model',
+                name: 'model',
+                type: 'BaseLanguageModel'
+            },
+            {
+                label: 'Document',
+                name: 'document',
+                type: 'Document',
+            }
+        ]
+    }
+
+    async init(nodeData: INodeData): Promise<any> {
+        const model = nodeData.inputs?.model as BaseLanguageModel
+        const docs = nodeData.inputs?.document as Document[]
+
+        const chain = await getOpenAPIChain(docs, model)
+        return chain
+    }
+
+    async run(nodeData: INodeData, input: string,  options: ICommonObject): Promise<string> {
+        const model = nodeData.inputs?.model as BaseLanguageModel
+        const docs = nodeData.inputs?.document as Document[]
+
+        const chain = await getOpenAPIChain(docs, model)
+        if (options.socketIO && options.socketIOClientId) {
+            const handler = new CustomChainHandler(options.socketIO, options.socketIOClientId)
+            const res = await chain.run(input, [handler])
+            return res
+        } else {
+            const res = await chain.run(input)
+            return res
+        }
+    }
+}
+
+const getOpenAPIChain = async (documents: Document[], llm: BaseLanguageModel, options: any = {}) => {
+    const texts = documents.map(({ pageContent }) => pageContent);
+    const apiResponsePrompt = new PromptTemplate({
```

Comment:
do you see an improvement from using custom `apiResponsePrompt` instead of the original prompt used in langchain library?

Reply:
@HenryHengZJ I was able to see less number tokens used and was getting meaningful results. May be we can give user an option to choose this and default one? what do you think ?

Reply:
ah okay i think its fine, just curious thats all. 

I'll make update once this PR is merged to get the POST API in. Thanks for making a head start!